### ISSUE TITLE
Pyinstaller can not run about Bio.Phylo.TreeConstruction

### ISSUE THREAD

**Issue by qianfeng2** [pyinstaller/pyinstaller#5732]
Dear whom it may concern,

My python scripts and related .exe using Pyinstaller used to work very well. Thanks very much for Pyinstaller and its team!!

When I add "**from Bio.Phylo.TreeConstruction import DistanceCalculator**" in my python script, it indeed builds a .exe successfully, but when I run this exe file, it returns an error, as below. Note that when I remove this sentence and its usage in my python script, .exe works okay again. 

**File not found error: [WinError3] The system cannot find the path specified: 'C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\_MEI36442\\Bio\\Align\\substitution_matrices\\data'[1664] Failed to execute script myscript**

However, I strongly need this DistanceCalculator which is from Biopython(v1.78) module, Is there anything I can do?

Thanks in advance.

**Comment by rokm** [comment]
If the package you're using comes with data files, you need to ensure that they are collected (PyInstaller doesn't do it automatically). You could [add the missing file](https://pyinstaller.readthedocs.io/en/stable/runtime-information.html#using-file) using `--add-data src;dst` command-line switch, but it seems `Bio.Align.substitution_matrices` has many of them, so it's probably easiest to write a hook for it:

* create directory called `extra-hooks` in your working directory
* create file `extra-hooks\hook-Bio.Align.substitution_matrices.py` with the following content:
```python
# hook-Bio.Align.substitution_matrices.py
from PyInstaller.utils.hooks import collect_data_files
datas = collect_data_files('Bio.Align.substitution_matrices')
```
* pass the `extra-hooks` directory to PyInstaller via `--additional-hooks` command-line switch, e.g.: `python -m PyInstaller --additional-hooks extra-hooks\ ...`.

**Comment by qianfeng2** [comment]
Hi @rokm 
I tried using your idea. It still not did not work, haha, I got the following error. But do not worry, I have solved my issue using raw code to replace DistanceCalculator of Biopython, since using hook seems a high level usage for me.

File not found error: [WinError3] The system cannot find the path specified: 'C:\Users\ADMINI~1\AppData\Local\Temp\_MEI36442\Bio\Align\substitution_matrices\data'[6588] Failed to execute script myscript

I will close this issue. Thank you again for your help!
